packages/web-components/fast-components/src/number-field/number-field.styles.ts

```diff
@@ -44,6 +45,7 @@ export const numberFieldStyles = (context, definition) =>
         .root {
             display: flex;
             flex-direction: row;
+            gap: calc(${densityComponentHorizontalBetweenPadding} * 1px);
```

Comment:
If we can make some changes so `gap` becomes meaningful, this would likely be a common pattern of setting `display: flex` and adding the correct `gap` for either `row` (horizontal) or `column` (vertical) layout.